backbone/src/main/java/org/researchstack/backbone/ui/ConsentTaskActivity.kt

```diff
@@ -0,0 +1,219 @@
+            val columnValue = String.format(signatureElementWrapper, completeName)
+            val column = arrayOf(columnValue, hr, columnName)
+            table.add(column)
+        }
+
+        if (signatureB64 != null) {
+            imageTag = "<img width='100%' alt='star' style='max-height:100px;max-width:200px;height:auto;width:auto;'" +
+                    " src='data:image/png;base64,$signatureB64'/>"
+            val columnName = getString(R.string.rsb_consent_doc_line_signature, role)
+            val columnValue = String.format(signatureElementWrapper, imageTag)
+            val column = arrayOf(columnValue, hr, columnName)
+            table.add(column)
+        }
+
+        if (table.size > 0) {
+            val columnName = getString(R.string.rsb_consent_doc_line_date)
+            val columnValue = String.format(signatureElementWrapper, signatureDate!!)
+            val column = arrayOf(columnValue, hr, columnName)
+            table.add(column)
+        }
+
+        if (table.size > 0) {
+            val columns = table[0].size
+            body.append("<table width='100%'>")
+            for (i in 0 until columns) {
+                body.append("<tr>")
+                for (j in table.indices) {
+                    body.append(td).append(table[j][i]).append("</td>")
+                }
+                body.append("</tr>")
+            }
+            body.append("</table>")
+        }
+
+        return body.toString()
+    }
+
+    internal inner class PDFWriteExposer : RSHTMLPDFWriter() {
+        public override fun printPdfFile(context: Activity, taskId: String?, htmlConsentDocument: String,
+                                         assetsFolder: String?, callback: PDFFileReadyCallback) {
+            super.printPdfFile(context, taskId, htmlConsentDocument, assetsFolder, callback)
+        }
+    }
+
+    companion object {
+
+        private const val ID_FORM_FIRST_NAME = "user_info_form_first_name"
+        private const val ID_FORM_LAST_NAME = "user_info_form_last_name"
+        private const val ID_FORM_DOB = "user_info_form_dob"
+        private const val ID_FORM = "user_info_form"
+        private const val EXTRA_ASSETS_FOLDER = "extra_assets_folder"
+
+        fun newIntent(context: Context, task: Task, assetsFolder: String): Intent {
+            val intent = Intent(context, ConsentTaskActivity::class.java)
+            intent.putExtra(EXTRA_TASK, task)
+            intent.putExtra(EXTRA_ASSETS_FOLDER, assetsFolder)
+            return intent
+        }
+
+        /**
+         * Returns the personal form steps related with the consent view task.
+         *
+         *
+         * This new version is prepared to work with multilanguage.
+         * Please use this method instead of
+         * [getConsentPersonalInfoFormStep][.getConsentPersonalInfoFormStep]
+         */
+        fun getConsentPersonalInfoFormStep(context: Context?, requiresName: Boolean,
```

Comment:
this static method... it's only used from RS* and I *do not like it at all*. If you can imagine a better way, go for it, otherwise, it's fine for now.

Reply:
I also don't like the nested `if`s and `context != null` checks all over the place, but it's fine for now.

Reply:
Moved to RSConsentConverter